### PR TITLE
Obey Map and Database save_count when loading games

### PR DESCRIPTION
**Pull request by fmatthew5876**
Resolves: #1203 

I tested this by creating a new project in Steam RM2k.

A map event with this code:
```
Text: before save
OpenSaveMenu
Text: After Save
```

Another map event which walks around randomly.

And a parallel common event with this code:
```
Text: Start
Loop
  Wait 4.0 seconds
EndLoop
```

This also correctly writes the database_save_count to the LSD file. Before it looks like it was never written to save games.

### CODE REVIEW

**Comment by fmatthew5876** [comment]
Added support for Map Chunk 0x5A.

I tested this with the following scenarios:
* rm2k steam game
* rm2k3 steam game converted from rm2k steam
* rm2k3 steam game converted from rm2k steam where the map was converted (saved) but the ldb was not. In this case rm2k3 doesn't reset the event states when it should and neither does player.

Now this depends on: https://github.com/EasyRPG/liblcf/pull/263

**Comment by fdelapena** [comment]
Jenkins: test this please

File: src/game_map.cpp [hunk 1]
```diff
@@ -258,9 +258,26 @@ void Game_Map::SetupCommon(int _id) {
 	pan_wait = false;
 	pan_speed = 0;
 
-	// Make RPG_RT happy
-	// Otherwise current event not resumed after loading
-	location.map_save_count = map->save_count;
+	auto map_save_count = map->save_count;
+	if (Data::system.ldb_id == 2003 && map->save_count_2k3e > 0) {
```
Comment (Ghabry) [inline src/game_map.cpp]: instead of ``Data::system.ldb_id == 2003`` use Player::IsRPG2k3

Comment (fmatthew5876) [reply src/game_map.cpp]: Fixed

**Pull request merged**